python/src/spark_rapids_ml/classification.py

```diff
@@ -985,11 +1013,27 @@ def _single_fit(init_parameters: Dict[str, Any]) -> Dict[str, Any]:
                     pdesc.rank,
                 )
 
+                intercept_array = logistic_regression.intercept_
+                # follow Spark to center the intercepts for multinomial classification
+                if (
+                    init_parameters["fit_intercept"] is True
+                    and len(intercept_array) > 1
+                ):
+                    import cupy as cp
+
+                    intercept_mean = (
+                        np.mean(intercept_array)
+                        if isinstance(intercept_array, np.ndarray)
+                        else cp.mean(intercept_array)
+                    )
```

Comment:
Doesn't this impact the model prediction ?

Reply:
Good question. It seems not affect the predicted class. Logistic is monotonically increasing and softmax predicts to the class that has the largest logistic value.

Reply:
It seems affect the rawPredict value but not max(rawPredict values)

Reply:
Reference in Spark code: https://github.com/apache/spark/blob/2a7e8900ab5f3845c3ebe6f43ce9b5ccd95cec99/mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala#L682

Reply:
I see.  It is only for multi-class and the soft-max is unaffected since each exponential is scaled by the same factor.